code/agents.py

```diff
@@ -64,13 +65,30 @@ def  __init__(self, position, velocity, fish_id, environment, ann, image = None,
     self._time_since_last_ANN_update
     self.fish_index
     """
-    
+
+    def get_fish_forces(self):
+        f = np.array([0.0, 0.0])
+        neighbours = self.get_neighbours()
+        k = self.environment.k
+        power = self.power
+        for fish in neighbours:
+            print("neihgbour detected at"+str(self.position))
+            f_i = k / np.abs(fish.position-self.position)**power
+            f += f_i
+
+        return f
+
+    def get_neighbours(self):
```

Comment:
Good move to have this in dedicated function, improves readability. But change name to find_neighbours, there are no getters in Python.